src/main/java/org/opensearch/ad/feature/CompositeRetriever.java

```diff
@@ -284,11 +303,28 @@ private boolean shouldRetryDueToEmptyPage(SearchResponse response) {
         }
 
         Optional<CompositeAggregation> getComposite(SearchResponse response) {
+            // When the source index is a regex like blah*, we will get empty response like
+            // the following even if no index starting with blah exists.
+            // {"took":0,"timed_out":false,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0},"hits":{"max_score":0.0,"hits":[]}}
+            // Without regex, we will get IndexNotFoundException instead.
+            // {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such
+            // index
+            // [blah]","index":"blah","resource.id":"blah","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no
+            // such index
+            // [blah]","index":"blah","resource.id":"blah","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}%
             if (response == null || response.getAggregations() == null) {
-                return Optional.empty();
+                List<String> sourceIndices = anomalyDetector.getIndices();
+                String[] concreteIndices = indexNameExpressionResolver
```

Comment:
Does this basically check if at least one of the indices in the pattern exists? Also This method won't get executed if detector has a single index used and it doesn't exist as an exception will be thrown earlier right?

Reply:
> Does this basically check if at least one of the indices in the pattern exists?

Yes

> Also This method won't get executed if detector has a single index used and it doesn't exist as an exception will be thrown earlier right?

For single index, you meant just index name without regex, right? If yes, right. We will get index not found exception before reaching here.

Reply:
Yeah meant just index name without regex.